src/Tasks/Microsoft.NET.Build.Tasks/PublishAssembliesResolver.cs

```diff
@@ -85,5 +101,14 @@ private static string GetRuntimeTargetDestinationSubDirectory(LockFileRuntimeTar
 
             return null;
         }
+
+        private string GetDestinationSubDirectory(string libraryPath, string pkgRoot, string destpath = null)
+        {
+            if (_preserveCacheLayout)
+            {
+                destpath = Path.GetDirectoryName(libraryPath.Replace(pkgRoot, ""));
```

Comment:
Do you want to replace all the occurences?  Or just the beginning?  Maybe checking if the libraryPath starts with pkgRoot, then substring the pkgRoot length off of libraryPath.